mwclient/client.py

```diff
@@ -877,25 +955,34 @@ def get_token(self, type, force=False, title=None):
 
         return self.tokens[type]
 
-    def upload(self, file=None, filename=None, description='', ignore=False,
-               file_size=None, url=None, filekey=None, comment=None):
+    def upload(
+        self,
+        file: Union[str, BinaryIO, TextIO, None] = None,
+        filename: Optional[str] = None,
+        description: str = '',
+        ignore: bool = False,
+        file_size: Optional[int] = None,
+        url: Optional[str] = None,
+        filekey: Optional[str] = None,
+        comment: Optional[str] = None
+    ) -> Any:
```

Comment:
I think the return value will always be an API response, we can use `Dict[str, Any]` here like we do for `api()`.

Reply:
We can, but it requires some casts:

```diff
diff --git a/mwclient/client.py b/mwclient/client.py
index abb3f5e..7e65dc3 100644
--- a/mwclient/client.py
+++ b/mwclient/client.py
@@ -965,7 +965,7 @@ class Site:
         url: Optional[str] = None,
         filekey: Optional[str] = None,
         comment: Optional[str] = None
-    ) -> Any:
+    ) -> Dict[str, Any]:
         """Upload a file to the site.
 
         Note that one of `file`, `filekey` and `url` must be specified, but not
@@ -1090,7 +1090,7 @@ class Site:
 
         if file is not None:
             file.close()
-        return response
+        return cast(Dict[str, Any], response)
 
     def chunk_upload(
         self,
@@ -1099,7 +1099,7 @@ class Site:
         ignorewarnings: bool,
         comment: str,
         text: Optional[str]
-    ) -> Any:
+    ) -> Dict[str, Any]:
         """Upload a file to the site in chunks.
 
         This method is called by `Site.upload` if you are connecting to a newer
@@ -1154,7 +1154,7 @@ class Site:
                 # get the parameters we need to continue, so we should return
                 # the response now.
                 file.close()
-                return response
+                return cast(Dict[str, Any], response)
 
         del params['action']
         del params['stash']
```